crates/fulf/src/interface/mod.rs

```diff
@@ -0,0 +1,637 @@
+    ///
+    /// 1. A mutable slice of unsorted results provided by fzy algorithm;
+    /// Those should be always sorted within the function
+    /// (but partially, as only 512 results are kept in the storage).
+    ///
+    /// 2. A number of total results that passed the matcher and provided
+    /// at least some score. The number of total results could be bigger than
+    /// the length of slice.
+    ///
+    /// # Returns
+    ///
+    /// Returns what `spawner` returns, but the type is defined by fzy algorithm.
+    ///
+    /// # Alternatives
+    ///
+    /// If you need a better control over algorithms, rules and directory
+    /// traversal, use `setter` function.
+    ///
+    /// If you need to read files in a manner different from `ignore::Walk`,
+    /// you can use `spawner` function.
+    ///
+    /// If you need something much different than anything there,
+    /// go and write it yourself.
+    #[inline]
+    pub fn default_searcher(
+        path: impl AsRef<Path>,
+        needle: impl Into<Box<str>>,
+        sort_and_print: impl FnMut(&mut [MWP], usize),
+    ) -> Option<(Vec<MWP>, usize)> {
+        with_fzy_algo(path, needle, 1024_usize.next_power_of_two(), sort_and_print)
+    }
+
+    /// A function to use default fuzzy-search algorithm.
+    ///
+    /// # Returns
+    ///
+    /// Return `None` if the root path cannot be represented as a utf8.
+    ///
+    /// # Maximum line length
+    ///
+    /// `max_line_len` sets maximum number of bytes for any line.
+    ///
+    /// If the line exceeds that number, it is not checked for match at all.
+    ///
+    /// Reasons:
+    ///
+    /// The speed of line-fuzzing is non-linear, thus lines too big
+    /// can slow down the task significantly. And there's very few reasons
+    /// for a line to exceed, for example, 1024 bytes:
+    ///
+    /// 1. This is a line in a text that is not code.
+    ///
+    /// 2. This is a non-formatted line of automatically generated code.
+    ///
+    /// 3. This is a very bad code.
+    ///
+    /// 4. Some very rare other reasons, like giant right-shifted branching.
+    ///
+    /// And in any of those cases there's probably no point in fuzzing such line.
+    #[inline]
+    pub fn with_fzy_algo(
+        path: impl AsRef<Path>,
+
+        needle: impl Into<Box<str>>,
+        max_line_len: usize,
+
+        sort_and_print: impl FnMut(&mut [MWP], usize),
+    ) -> Option<(Vec<MWP>, usize)> {
+        let needle = needle.into();
+
+        if needle.is_empty() || needle.len() > max_line_len {
+            return Default::default();
+        }
+
+        let r = {
+            let mut r = Rules::new();
+            r.bonus_threads = if cfg!(target_pointer_width = "64") {
+                2
+            } else {
+                0
+            };
+
+            r
+        };
+
+        let path = path.as_ref();
+        let dir_iter = ignore::Walk::new(path);
+
+        let root_folder = path.to_str()?;
```

Comment:
In that line the whole thing could fail because:

First: if you use `OsStr::to_string_lossy()`, you cannot restore original path.
Second: there is no easy way to represent the path as raw bytes.

Thus it's easier to ignore any path that can't be represented as utf8. Anyway, no sane person would store anything useful in a path that could not be represented as Unicode.